x-pack/plugins/cases/public/components/all_cases/assignees_column.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+import { EuiButtonEmpty, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import type { UserProfileWithAvatar } from '@kbn/user-profile-components';
+import type { Case } from '../../../common/ui/types';
+import { getEmptyTagValue } from '../empty_value';
+import { UserToolTip } from '../user_profiles/user_tooltip';
+import { useAssignees } from '../../containers/user_profiles/use_assignees';
+import { getUsernameDataTestSubj } from '../user_profiles/data_test_subject';
+import { SmallUserAvatar } from '../user_profiles/small_user_avatar';
+import * as i18n from './translations';
+
+const COMPRESSED_AVATAR_LIMIT = 5;
+
+export interface AssigneesColumnProps {
+  assignees: Case['assignees'];
+  userProfiles: Map<string, UserProfileWithAvatar>;
+  compressedDisplayLimit?: number;
+}
+
+const AssigneesColumnComponent: React.FC<AssigneesColumnProps> = ({
+  assignees,
+  userProfiles,
+  compressedDisplayLimit = COMPRESSED_AVATAR_LIMIT,
+}) => {
+  const [isAvatarListExpanded, setIsAvatarListExpanded] = useState<boolean>(false);
+
+  const { allAssignees } = useAssignees({
+    caseAssignees: assignees,
+    userProfiles,
+  });
+
+  const toggleExpandedAvatars = useCallback(
+    () => setIsAvatarListExpanded((prevState) => !prevState),
+    []
+  );
+
+  const numHiddenAvatars = allAssignees.length - compressedDisplayLimit;
+  const shouldShowExpandListButton = numHiddenAvatars > 0;
+
+  const avatarsToDisplay = useMemo(() => {
+    if (isAvatarListExpanded || !shouldShowExpandListButton) {
+      return allAssignees;
+    }
+
+    return allAssignees.slice(0, compressedDisplayLimit);
```

Comment:
nit: if we expected the `compressedDisplayLimit` to not change during the lifecycle of the component maybe we can move out the `slice` out of this `useMemo` and put it on another one that depends only on the `compressedDisplayLimit`. The reason is to avoid slicing every time the user presses the "Show more" button. Small optimization without measurements so maybe it is not worth it.